### PR TITLE
Add date transform

### PR DESCRIPTION
**Pull request by Casey1449**
@stevekinney @martensonbj @brittanystoroz 

## Purpose

Addresses the all requested changes from the previous two un-merged PRs, and adds edit functionality, record saving, more styling, and date formatting.

## Approach

We added an 'edit' route that basically mimics what the 'new' route does, but it uses the id param to open the route relative to a specific reminder, and auto-fills the 'new-reminder' form with the reminder's title and notes. 

### Learning

Lots. We moved the action to the edit button instead of the form, so that we could also nest it in a link helper, so we can perform the save action and the re-route all on the button click. We installed ember-moment for date formatting, and a transform to convert back and forth from what the form field (type=date) wants, to what we want to display, to what the mirage DB wants. Another big learning point was to let Ember do the work. After much struggle, we discovered that the two-way binding in handlebars inputs took care of all the 'grabbing' of info from the DOM, and we just had to save() it. 

### Test coverage 

To come. 

### Merge Dependencies and Related Work

We closed our previous 2 pull requests because we're getting issues assigned far faster than we're getting PRs merged, so we're working ahead while waiting for merges/comments, and we're addressing comments on future branches instead of going back to update outdated ones for merging. Is this right?

### CODE REVIEW

**Comment by brittanystoroz** [comment]
This is a pretty big PR. I recognize that you were trying to continue working while waiting on your PRs getting reviewed and merged. In the workplace, I handle similar situations in the following way:

* Create a new branch `feature-a` and work on feature A
* Submit a PR for feature A
* Feature B depends on the work you did for feature A
* The PR for feature A has not been merged yet, but you want to continue making progress
* Create a new branch `wip-feature-b`, based off of your `feature-a` branch
* Finish work for feature B
* Wait for feature A to be merged into master
* Create a new branch `feature-b` based off of master, which now has feature A's code
* Use `[git cherry-pick](https://git-scm.com/docs/git-cherry-pick) <commit sha>` to pull in **just** the commits you made in `wip-feature-b` that are specifically for feature B
* Submit a PR for feature B with the `feature-b` branch

@steve might know an easier, more sane way to do this, but that's been my strategy and it's worked out pretty well.

File: app/routes/reminder-list/new.js [hunk 1]
```diff
@@ -0,0 +1,4 @@
+import Ember from 'ember';
+
+export default Ember.Route.extend({
+});
```
Comment (brittanystoroz) [inline app/routes/reminder-list/new.js]: Is this file doing something in particular? Is it needed?

File: app/styles/app.scss [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+
+p {
+  font-family: $serif;
+  font-size: 1.1em;
+}
+
+h2 {
+  margin-top:0px;
+  border-bottom: 1px solid $dark-accent-color;
+
```
Comment (brittanystoroz) [inline app/styles/app.scss]: Get rid of this extra blank line

File: app/styles/app.scss [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+.reminders-list-container {
+  display: flex;
+  flex-direction: column;
+  justify-content: center;
+  background-color: $primary-color-1;
+}
+
+.reminder-list {
+  height: 80vh;
+
```
Comment (brittanystoroz) [inline app/styles/app.scss]: This blank line too.

File: app/styles/app.scss [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+  font-size: 14px;
+}
+
+.reminder-submit-button {
+  padding-top: 5px;
+  font-size: 25px;
+}
+
+
+form{
```
Comment (brittanystoroz) [inline app/styles/app.scss]: Add a space before this brace

File: app/styles/app.scss [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+  background-color: $light-accent-color;
+  color: $dark-accent-color;
+  padding: 20px;
+  border-radius: 10px;
+  border: 2px solid $dark-accent-color;
+  max-width: 70%;
+  margin: auto;
+}
+
+input{
```
Comment (brittanystoroz) [inline app/styles/app.scss]: And here

File: app/styles/app.scss [hunk 6]
```diff
@@ -0,0 +1,179 @@
...
+  justify-content: center;
+}
+
+.reminder-list-item {
+  padding: 15px;
+  margin: 0px;
+  border-radius: 5px;
+}
+
+.reminder-list-item:hover{
```
Comment (brittanystoroz) [inline app/styles/app.scss]: Let's be consistent with adding a space before these braces.

File: app/styles/app.scss [hunk 7]
```diff
@@ -0,0 +1,179 @@
...
+  margin: 0px;
+}
+
+.reminder-item-date{
+  font-family: $sans;
+  margin: 0px
+}
+
+.reminder-list-item.active {
+  border:none;
```
Comment (brittanystoroz) [inline app/styles/app.scss]: spaces after the colon as well

File: app/templates/reminder-list/reminder-detail.hbs [hunk 8]
```diff
@@ -1,7 +1,10 @@
-<article>
+<article class="reminder-item-detail">
   {{#with model as |reminder| }}
+  {{#link-to "reminder-list.edit" reminder}}
+  <span title='edit this reminder'><button class="edit-button">📝</button></span>
```
Comment (brittanystoroz) [inline app/templates/reminder-list/reminder-detail.hbs]: Let's be consistent with using either single quotes or double quotes

File: app/transforms/date.js [hunk 9]
```diff
@@ -0,0 +1,14 @@
+import DS from 'ember-data';
+import moment from 'moment';
+
+export default DS.Transform.extend({
+  deserialize(serialized) {
+    if (serialized){return moment(serialized).format("dddd, MMM Do, h:mm a");}
```
Comment (brittanystoroz) [inline app/transforms/date.js]: This code looks sooo suffocated. Can we spread it out onto another line?

File: app/styles/app.scss (hunk 3 above)
Comment (Casey1449) [inline app/styles/app.scss]: ✔️

File: app/routes/reminder-list/new.js (hunk 1 above)
Comment (Casey1449) [inline app/routes/reminder-list/new.js]: nope! I think it just came along for the ride when we ember g-ed something. deleted.

File: app/styles/app.scss (hunk 4 above)
Comment (Casey1449) [inline app/styles/app.scss]: ✔️

File: app/styles/app.scss (hunk 5 above)
Comment (Casey1449) [inline app/styles/app.scss]: ✔️

File: app/styles/app.scss (hunk 6 above)
Comment (Casey1449) [inline app/styles/app.scss]: Ok, I think I got 'em all.

File: app/styles/app.scss (hunk 7 above)
Comment (Casey1449) [inline app/styles/app.scss]: ✔️

File: app/templates/reminder-list/reminder-detail.hbs (hunk 8 above)
Comment (Casey1449) [inline app/templates/reminder-list/reminder-detail.hbs]: Ah, shoot. Usually my jshint catches that, but I think the linter is overriding it or something? I think I got em all in the hbs files.

File: app/transforms/date.js (hunk 9 above)
Comment (Casey1449) [inline app/transforms/date.js]: Yup. Early in mod one, they instructors told us they preferred one-line 'if' statements...I'll try to break the habit.

Comment (brittanystoroz) [inline app/transforms/date.js]: 😮

**Pull request merged**